From ed9a23dd10e2e8f0b682f95b195d64ec019d7f04 Mon Sep 17 00:00:00 2001
From: Kirill Kapranov <kirill.kapranov@compulab.co.il>
Date: Sun, 3 Oct 2021 01:41:43 +0300
Subject: [PATCH 094/126] mmc: Fix mmc_switch workflow issue

Revert the patch 9a461acedebe664cf1e557c3b9ec983e6b04e805, that introduces
premature falling back on a long timeout instead of continuation of the
initialization.
Clear the ambiguity of the comment that led this.

Signed-off-by: Kirill Kapranov <kirill.kapranov@compulab.co.il>
---
 drivers/mmc/mmc.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/mmc/mmc.c b/drivers/mmc/mmc.c
index 785ebc0cd3..6d9766a697 100644
--- a/drivers/mmc/mmc.c
+++ b/drivers/mmc/mmc.c
@@ -819,11 +819,11 @@ static int __mmc_switch(struct mmc *mmc, u8 set, u8 index, u8 value,
 		return ret;
 
 	/*
-	 * In cases when not allowed to poll by using CMD13 or because we aren't
+	 * In cases we aren't allowed to poll by using CMD13, nor are we
 	 * capable of polling by using mmc_wait_dat0, then rely on waiting the
 	 * stated timeout to be sufficient.
 	 */
-	if (ret == -ENOSYS || !send_status) {
+	if (ret == -ENOSYS && !send_status) {
 		mdelay(timeout_ms);
 		return 0;
 	}
-- 
2.11.0

